reana_workflow_controller/rest/utils.py

```diff
@@ -180,27 +180,32 @@ def delete_workflow(workflow,
                            WorkflowStatus.finished,
                            WorkflowStatus.stopped,
                            WorkflowStatus.deleted,
-                           WorkflowStatus.failed]:
-        to_be_deleted = [workflow]
-        if all_runs:
-            to_be_deleted += Session.query(Workflow).\
-                filter(Workflow.name == workflow.name,
-                       Workflow.status != WorkflowStatus.running).all()
-        for workflow in to_be_deleted:
-            if hard_delete:
-                remove_workflow_workspace(workflow.workspace_path)
-                _delete_workflow_row_from_db(workflow)
-            else:
-                if workspace:
+                           WorkflowStatus.failed,
```

Comment:
Could you confirm what is happening with the [`reana-server` `scheduler` container](https://github.com/reanahub/reana-server/blob/fb00c8b049b0871a63fc46d9bddae9dc280052ad/reana_server/scheduler.py#L79)? Because we have this operation to delete from the DB, but the workflow will still be present in the RabbitMQ queue... (which one can check with `rabbitmqctl list_queues` inside the `message-broker` component).

I am afraid that `reana-server` will continuously try to start the workflow, fail, requeue, try to start, fail, requeue... infinitely.

If that is the case, I feel we need to find a nice way of doing this operation atomically.

Reply:
This [PR](https://github.com/reanahub/reana-server/pull/214) is solving this.